docs/how-tos/rust_service.md

```diff
@@ -0,0 +1,347 @@
+    preferred_name: Optional<String>,
+}
+```
+We can write a resolver for the `Person`. A resolver function resolvers the values for all the fields a GraphQL object.
+
+!!! example "Defind Object and a Resolver"
+
+    ```rust
+    use async_graphql::Object;
+    struct Query;
+
+    #[Object]
+    impl Query {
+        async fn person(&self) -> Person {
+            Person {
+                id: 1, 
+                first_name: "foo".to_string(), 
+                last_name: "bar".to_string(),
+                preferred_name: None,
+            }
+        }
+    }
+    ```
+Putting everything together, we should be able to run the follwing code, 
+
+!!! example "Simple GraphQL service to provide a Person information"
+
+    ```rust
+    use async_graphql::{Context, EmptyMutation, EmptySubscription, Object, Schema, SimpleObject};
+    use async_graphql_axum::{GraphQLRequest, GraphQLResponse};
+    use axum::{
+        extract::Extension,
+        routing::post,
+        Router,
+    };
+
+    struct Query;
+
+    #[Object]
+    impl Query {
+        async fn person(&self) -> Person {
+            Person {
+                id: 1, 
+                first_name: "foo".to_string(), 
+                last_name: "bar".to_string(),
+                preferred_name: None,
+            }
+        }
+    }
+
+    #[derive(SimpleObject)]
+    struct Person {
+        id: u32, 
+        first_name: String, 
+        last_name: String, 
+        preferred_name: Option<String>,
+    }
+
+    #[tokio::main]
+    async fn main() {
+        let schema = Schema::build(Query, EmptyMutation, EmptySubscription).finish();
+
+        let app = Router::new()
+            .route("/graphql", post(graphql_handler))
+            .layer(Extension(schema));
+
+        let listener = tokio::net::TcpListener::bind("127.0.0.1:8000").await.unwrap();
+        axum::serve(listener, app).await.unwrap();
+    }
+
+    async fn graphql_handler(
+        schema: Extension<Schema<Query, EmptyMutation, EmptySubscription>>,
+        req: GraphQLRequest,
+    ) -> GraphQLResponse {
+        schema.execute(req.into_inner()).await.into()
+    }
+    ```
+When we send request to the endpoint `curl -X POST -H "Content-Type: application/json" -d '{"query":"{ person {id, firstName, lastName, preferredName} }"}' http://127.0.0.1:8000/graphql` we should get the following response
+```json
+{
+    "data":{
+        "person":{
+            "id":1,
+            "firstName":"foo",
+            "lastName":"bar",
+            "preferredName":null
+            }
+        }
+}
+```
+
+### Related
+
+Most of the fields of a GraphQL object directly return the value of the field,
+but sometimes the the fields are calculated or resolved by a different resolver.
+In such cases, we can use the `ComplexObject` macro to write a user defined resolver.
+This can be useful when we want to resolve related Objects. 
+We need to use `complex` macro on `Person` struct for the `ComplexObject` macro to take effect.
```

Comment:
Would be good to mention where we are using the macros (i.e. on the `struct` or `impl`)